standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -3475,15 +3496,28 @@ public List<Partition> listPartitionsPsWithAuth(String catName, String db_name,
 
     try {
       openTransaction();
+
+      MTable mtbl = getMTable(catName, db_name, tbl_name);
+      if (mtbl == null) {
+        throw new NoSuchObjectException(
+            TableName.getQualified(catName, db_name, tbl_name) + " table not found");
+      }
+      boolean getauth = null != userName && null != groupNames &&
+          "TRUE".equalsIgnoreCase(
+              mtbl.getParameters().get("PARTITION_LEVEL_PRIVILEGE"));
+      if (!getauth && canTryDirectSQL(part_vals)) {
+        LOG.info(
+            "Redirecting to directSQL enabled API: db: {} tbl: {} partVals: {}",
+            db_name, tbl_name, Joiner.on(',').join(part_vals));
+        return getPartitions(catName, db_name, tbl_name, -1);
```

Comment:
Hi @vihangk1 Why change max_parts to -1?  Can we change this line to:
```java
return getPartitions(catName, db_name, tbl_name, max_parts);
```

Reply:
yeah, thats a good catch. We should fix it. Can you create a PR and I can review it.

Reply:
https://github.com/apache/hive/pull/4662

Reply:
I see @ayushtkn already has approved it. I will let him merge it.